Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMDecryptorGStreamer.cpp

```diff
@@ -85,14 +87,27 @@ static GRefPtr<GstCaps> createSinkPadTemplateCaps()
     std::string emptyString;
     GRefPtr<GstCaps> caps = adoptGRef(gst_caps_new_empty());
 
-    if (!opencdm_is_type_supported(openCDMAccessor.get(), WebCore::GStreamerEMEUtilities::s_ClearKeyKeySystem, emptyString.c_str()))
+    bool supportsEME = false;
+    if (!opencdm_is_type_supported(openCDMAccessor.get(), WebCore::GStreamerEMEUtilities::s_ClearKeyKeySystem, emptyString.c_str())) {
         addKeySystemToSinkPadCaps(caps, WEBCORE_GSTREAMER_EME_UTILITIES_CLEARKEY_UUID);
+        supportsEME = true;
+    }
 
-    if (!opencdm_is_type_supported(openCDMAccessor.get(), WebCore::GStreamerEMEUtilities::s_PlayReadyKeySystems[0], emptyString.c_str()))
+    if (!opencdm_is_type_supported(openCDMAccessor.get(), WebCore::GStreamerEMEUtilities::s_PlayReadyKeySystems[0], emptyString.c_str())) {
         addKeySystemToSinkPadCaps(caps, WEBCORE_GSTREAMER_EME_UTILITIES_PLAYREADY_UUID);
+        supportsEME = true;
+    }
 
-    if (!opencdm_is_type_supported(openCDMAccessor.get(), WebCore::GStreamerEMEUtilities::s_WidevineKeySystem, emptyString.c_str()))
+    if (!opencdm_is_type_supported(openCDMAccessor.get(), WebCore::GStreamerEMEUtilities::s_WidevineKeySystem, emptyString.c_str())) {
         addKeySystemToSinkPadCaps(caps, WEBCORE_GSTREAMER_EME_UTILITIES_WIDEVINE_UUID);
+        supportsEME = true;
+    }
+
+    // No key system UUD for webm. It's not set in caps for it.
+    if (supportsEME) {
+        for (int i = 0; webmEncryptionMediaTypes[i]; ++i)
+            gst_caps_append_structure(caps.get(), gst_structure_new("application/x-webm-enc", "original-media-type", G_TYPE_STRING, webmEncryptionMediaTypes[i], nullptr));
+    }
```

Comment:
I feel this is wrong. I think you should leave the PlayReady and ClearKey code as it was and then, if Widevine is supported, you can also assume that WebM/Widevine is supported and add the caps for it. `supportsEME` is not needed (and if it were, it should be called `doesSupportEME`).

Reply:
The point of thi sis not to set protection system UUID as it's not there in the caps which comes in

Reply:
NVM. I misread the comment. Fix comming in.